src/control/lib/daos/container_property.go

```diff
@@ -0,0 +1,1214 @@
+	case ContainerPropStatus:
+		return C.DAOS_PROP_ENTRY_STATUS
+	case ContainerPropHighestOid:
+		return C.DAOS_PROP_ENTRY_ALLOCED_OID
+	case ContainerPropEcCellSize:
+		return C.DAOS_PROP_ENTRY_EC_CELL_SZ
+	case ContainerPropEcPerfDom:
+		return C.DAOS_PROP_ENTRY_EC_PDA
+	case ContainerPropEcPerfDomAff:
+		return C.DAOS_PROP_ENTRY_RP_PDA
+	case ContainerPropGlobalVersion:
+		return C.DAOS_PROP_ENTRY_GLOBAL_VERSION
+	case ContainerPropScubberDisabled:
+		return C.DAOS_PROP_ENTRY_SCRUB_DISABLED
+	case ContainerPropObjectVersion:
+		return C.DAOS_PROP_ENTRY_OBJ_VERSION
+	case ContainerPropPerfDomain:
+		return C.DAOS_PROP_ENTRY_PERF_DOMAIN
+	default:
+		return fmt.Sprintf("unknown container property type %d", cpt)
+	}
+}
+
+// FromString attempts to resolve the supplied string into a container property
+// type.
+func (cpt *ContainerPropType) FromString(in string) error {
+	// NB: The implication here is that every container property should
+	// have an entry in the handler map in order to be resolvable. This
+	// should be reasonable. The alternative is a manually-maintained
+	// list that is the inverse of String().
+	ph, err := propHdlrs.get(in)
+	if err != nil {
+		return err
+	}
+
+	*cpt = ContainerPropType(ph.dpeType)
+	return nil
+}
+
+// propHdlrs defines a map of property names to handlers that
+// take care of parsing the value and setting it. This odd construction
+// allows us to maintain a type-safe set of valid property names and
+// string properties in one place, and also enables user-friendly
+// command completion.
+//
+// Most new features requiring a change to the property handling
+// should only need to modify this map. If your new property has a
+// custom input processor or pretty-printer, please add a unit test
+// for that code!
+//
+// The structure of an entry is as follows:
+//
+//	"key": {                                    // used as the --property name
+//	        C.DAOS_PROP_ENUM_VAL,               // identify the property type
+//	        "short description",                // human-readable (short) description
+//	        closure of type entryHdlr,          // process user input (optional for read-only or mapped inputs)
+//	        map[string]valHdlr{                 // optional map of fixed-string input processors
+//	            "key": closure of type valHdlr, // convert fixed-string input to property value
+//	        },
+//	        []string,                           // optional slice of deprecated keys
+//	        closure of type entryStringer,      // optional pretty-printer
+//	        bool,                               // if true, property is read-only
+//	},
+var propHdlrs = propHdlrMap{
+	C.DAOS_PROP_ENTRY_LABEL: {
+		C.DAOS_PROP_CO_LABEL,
+		"Label",
+		func(_ *propHdlr, p *ContainerProperty, v string) error {
+			if !LabelIsValid(v) {
+				return errors.Errorf("invalid %s %q", p.Name, v)
+			}
+			p.entry.dpe_type = C.DAOS_PROP_CO_LABEL
+			return p.SetString(v)
+		},
+		nil, nil,
```

Comment:
nit: for this specific "dev-editable" map, I think it would be more readable to have each entry field on its own line

Reply:
OK